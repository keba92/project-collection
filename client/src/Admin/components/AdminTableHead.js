import React from 'react';
import { useTranslation } from 'react-i18next';

export default function AdminTableHead(props) {
    const { selectAll, checkedAll } = props;
    const { t, i18n } = useTranslation();

    return (
        <tr>
            <th>Check All<br/>
                <input type='checkbox' className ='form-check-input' 
                       onChange={(event) => selectAll(event.target.checked)} 
                       checked={checkedAll}/>
            </th>
            <th>ID</th>
            <th>{t('nameT')}</th>
            <th>{t('emailT')}</th>
            <th>{t('dateRegT')}</th>
            <th>{t('dateLastT')}</th>
            <th>{t('statusT')}</th>
            <th>{t('adminT')}</th>
        </tr>
    )
}