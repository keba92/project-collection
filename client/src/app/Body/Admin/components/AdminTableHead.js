import React, { memo } from 'react';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';

function AdminTableHead(props) {
  const { selectAll, checkedAll } = props;
  const { t, i18n } = useTranslation();

  return (
    <tr>
      <th>
        Check All
        <br />
        <input
          type="checkbox"
          className="form-check-input"
          onChange={(event) => selectAll(event.target.checked)}
          checked={checkedAll}
        />
      </th>
      <th>ID</th>
      <th>{t('nameT')}</th>
      <th>{t('emailT')}</th>
      <th>{t('dateRegT')}</th>
      <th>{t('dateLastT')}</th>
      <th>{t('statusT')}</th>
      <th>{t('adminT')}</th>
    </tr>
  );
}

AdminTableHead.propTypes = {
  selectAll: PropTypes.func,
  checkedAll: PropTypes.bool,
};

export default memo(AdminTableHead);
