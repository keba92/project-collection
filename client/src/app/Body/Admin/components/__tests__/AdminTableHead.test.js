import React from 'react';
import AdminTableHead from '../AdminTableHead';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

afterEach(cleanup);

const testFn = () => {
  return 'test';
};

it('matches snapshot', () => {
  const tree = renderer.create(<AdminTableHead selectAll={testFn} checkedAll={testFn} />).toJSON();
  expect(tree).toMatchSnapshot();
});
