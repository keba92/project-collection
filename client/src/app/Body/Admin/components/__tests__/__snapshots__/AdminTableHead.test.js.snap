// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<tr>
  <th>
    Check All
    <br />
    <input
      checked={[Function]}
      className="form-check-input"
      onChange={[Function]}
      type="checkbox"
    />
  </th>
  <th>
    ID
  </th>
  <th>
    nameT
  </th>
  <th>
    emailT
  </th>
  <th>
    dateRegT
  </th>
  <th>
    dateLastT
  </th>
  <th>
    statusT
  </th>
  <th>
    adminT
  </th>
</tr>
`;
